### PR TITLE
Add typing for steps/discover/shell.py

### PR DESCRIPTION
**Pull request by janhavlin**
Blocked until #1333 is merged.

### CODE REVIEW

**Comment by thrix** [comment]
@janhavlin this is now unblocked, pls rebase and continue with getting this into reviewable state

**Comment by janhavlin** [comment]
This should now be ready for review.

**Comment by thrix** [comment]
Moving to 1.17

File: tmt/steps/discover/shell.py [hunk 1]
```diff
@@ -42,32 +46,34 @@ class DiscoverShell(tmt.steps.discover.DiscoverPlugin):
...
         """ Wake up the plugin, process data, apply options """
         super().wake(keys=keys)
         # Check provided tests, default to an empty list
         if 'tests' not in self.data:
             self.data['tests'] = []
-        self._tests = []
+        self._tests: List[tmt.base.Test] = []
 
-    def go(self):
+    def go(self, *args: Any, **kwargs: Any) -> None:
```
Comment (happz) [inline tmt/steps/discover/shell.py:65]: Are `*args` and `**kwargs` needed? `DiscoverShell` inherits from `DiscoverPlugin` whose `def go(self) -> None` accepts no arguments anyway. (See https://github.com/teemtee/tmt/commit/b371a0d0d04c3e16384f3f1749f5b08efb2c7097 for the related refactoring).

Comment (janhavlin) [reply tmt/steps/discover/shell.py]: Right. This is something I forgot to do after the `go()` refactoring. Fixed, thanks.

**Review by psss — APPROVED** [review]
lgtm

**Review by thrix — CHANGES_REQUESTED** [review]
Sorry forgot to submit this comment

**Review by thrix — APPROVED** [review]
LGTM

**Comment by thrix** [comment]
review karma reached, rebased and preparing for merge

**Pull request merged**